141-health-endpoint/proposal.md

```diff
@@ -0,0 +1,56 @@
+# Reference
+* Original RFC [#46](https://github.com/concourse/rfcs/pull/46)
+* Original POC [#4818](https://github.com/concourse/concourse/pull/4818)
+
+# Summary
+This proposal outlines the beginnings of support for a `health` endpoint, which has a simple backend service which monitors crucial Concourse interfaces such as the database connectivity, the worker count (of healthy workers (*should be some threshold depending on the update strategy*)), the state of the webs (ATC/TSA) and others.
+
+# Motivation
+#### Currently, Concourse does not expose a dedicated, standardized health endpoint that external systems can query to determine the system’s overall health. This creates challenges in the following areas:
+
+### 1. Monitoring & Alerting
+Operators and platform teams often integrate Concourse with monitoring systems (e.g. Prometheus, Datadog, Kubernetes liveness/readiness probes). Without a clear health endpoint, they must rely on indirect signals (such as API responses, metrics, or manual checks), which can be unreliable or difficult to standardize.
+
+### 2. Automation & Self-Healing
+Modern infrastructure frequently depends on health endpoints for automated actions like restarting unhealthy pods, removing failing nodes from load balancers, or scaling workloads. The lack of a health endpoint makes such automation harder to implement for Concourse.
+
+### 3. User Experience
+When Concourse becomes partially degraded (e.g. workers are down, ATC is unresponsive, DB is lagging), it is not immediately obvious to users or operators. A health endpoint would provide a quick, single source of truth for identifying issues.
+
+### 4. Consistency with Industry Standards
+Most modern distributed systems (e.g. Kubernetes components, CI/CD systems, databases) expose health endpoints (commonly `/healthz`, `/readyz`, `/livez`). Introducing a similar endpoint in Concourse aligns it with best practices and user expectations.
+
+## What will it bring?
+By introducing a health endpoint, we make it easier to operate Concourse reliably in production environments, reduce the burden on operators, and enable better integration with external observability and orchestration systems.
+
+# Proposal
+## API Changes
+What comes to mind is a simple **unauthenticated** HTTP endpoint (e.g. `/health`) that returns a JSON payload indicating the overall health status of the Concourse system. Could be something simple like:
+```json
+{
+  "status": "healthy/unhealthy",
+  "details": {
+    "database": "healthy/unhealthy",
+    "workers": "healthy/unhealthy",
```

Comment:
"workers" as a single entry does not convey much meaningful information IMO. A list of the status of each worker might be more useful.  Also, the semantics of the general "status" should be clarified. What is considered a healthy instance  ?